server/routes/overview/viewGoalsView.ts

```diff
@@ -1,14 +1,21 @@
-import type { PrisonerSummary } from 'viewModels'
+import type { Goal, PrisonerSummary } from 'viewModels'
 
 export default class ViewGoalsView {
-  constructor(private readonly prisonerSummary: PrisonerSummary) {}
+  constructor(
+    private readonly prisonerSummary: PrisonerSummary,
+    private readonly inProgressGoals: Array<Goal>,
+    private readonly archivedGoals: Array<Goal>,
+    private readonly completedGoals: Array<Goal>,
+    private readonly problemRetrievingData: boolean,
+  ) {}
```

Comment:
I think you can tidy your constructor here, which will make your controller code neater.
Basically your constructor only needs to take the prisonerSummary and allGoalsForPrisoner: PrisonerGoals
```
  constructor(
    private readonly prisonerSummary: PrisonerSummary,
    private readonly allGoalsForPrisoner: PrisonerGoals,
  ) {}
```

Then your controller code becomes really neat and concise:
```
      const view = new ViewGoalsView(
        prisonerSummary,
        res.locals.allGoalsForPrisoner,
      )

      res.render('pages/overview/partials/goalsTab/goalsTabContents', view.renderArgs)
```

Then, your renderArgs method here needs to define it's return type as being different to the constructor, and then break out the fields from the PrisonerGoals:
```
  get renderArgs(): {
    tab: string
    prisonerSummary: PrisonerSummary
    problemRetrievingData: boolean
    inProgressGoals: Array<Goal>
    archivedGoals: Array<Goal>
    completedGoals: Array<Goal>
  } {
    return {
      tab: 'goals',      
      prisonerSummary: this.prisonerSummary,
      problemRetrievingData: this.allGoalsForPrisoner.problemRetrievingData,
      inProgressGoals: this.allGoalsForPrisoner.inProgressGoals,
      archivedGoals: this.allGoalsForPrisoner.archivedGoals,
      completedGoals: this.allGoalsForPrisoner.completedGoals,
    }
```